From e937a6ee99d95aadb2a3f24d75cb8db9f9e992da Mon Sep 17 00:00:00 2001
From: Vitaly Margolin <vmargolin@habana.ai>
Date: Tue, 27 May 2025 13:40:22 +0300
Subject: [PATCH] [SW-230361] g3: serdes: skip serdes part. validation

Validate SerDes LFS partition SHA3-384 checksum on
secured setups only and skip validation in non-secured
setups

Change-Id: I30f0c2a7303214a7463f08d379f8c6bae25d398c
Signed-off-by: Vitaly Margolin <vmargolin@habana.ai>
---

diff --git a/src/hl_serdes/gaudi3/serdes.c b/src/hl_serdes/gaudi3/serdes.c
index d4f702a..2fab0f5 100644
--- a/src/hl_serdes/gaudi3/serdes.c
+++ b/src/hl_serdes/gaudi3/serdes.c
@@ -17,6 +17,7 @@
 #include <cpucp_if.h>
 #include <zephyr/logging/log.h>
 #include <hl_devs.h>
+#include <hl_security.h>
 #include <arcboot_export_dev.h>
 #include <zephyr/drivers/misc/hl_sfp/hl_sfp.h>
 #include <zephyr/drivers/misc/hl_cpld/hl_cpld.h>
@@ -177,9 +178,12 @@
 
 static int start_lfs_serdes(void)
 {
-	int rc;
+	const struct device *security = SECURITY_DEV_INST;
+	int rc = 0;
 
-	rc = validate_lfs_serdes();
+	if (hl_security_is_enabled(security))
+		rc = validate_lfs_serdes();
+
 	if (!rc)
 		rc = lfs_zephyrbd_init(&zephyrbd_state, LFS_SPI_SERDES_PARTITION);
 
